Bottom: 2bc6cb096178e372d949d4049bbad45f93a2d518
Top:    f2b3a3703a86a2bb5deb792cffa742476d4d589f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-01-25 21:28:11 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 88afa1b..2bbff2a 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -309,6 +309,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
